driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -558,6 +563,10 @@ private static void refreshNodeListAndTokenMap(Connection connection, Cluster.Ma
             allTokens.add(row.getSet("tokens", String.class));
         }
 
+        // shuffle the list of found hosts to
+        // spread the load evenly
+        Collections.shuffle(foundHosts);
```

Comment:
I don't think shuffling makes any difference here: the elements of `foundHosts` are eventually added to `Cluster.metadata.hosts`, which is a concurrent hash map.
We'll have to rely on load balancing policies to properly randomize their query plans.